ext/ffi/lib.rs

```diff
@@ -768,6 +770,28 @@ fn is_i64(rv: NativeType) -> bool {
   matches!(rv, NativeType::I64 | NativeType::ISize)
 }
 
+#[cfg(not(target_os = "windows"))]
+fn is_fastapi_available() -> &'static bool {
+  #[cfg(target_os = "linux")]
+  static FASTAPI_IS_AVAILABLE: OnceCell<bool> = OnceCell::new();
+  #[cfg(target_os = "linux")]
+  return FASTAPI_IS_AVAILABLE.get_or_init(|| {
+    // SAFETY: mprotect should fail on malloc-ed memory on SELinux which prevent tinycc from working
+    unsafe {
+      let ptr = libc::malloc(1);
+      if ptr.is_null() {
```

Comment:
```suggestion
      if !ptr.is_null() {
```
It should call mprotect only if the ptr is not null